src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/MetricsStore.cs

```diff
@@ -95,6 +95,9 @@ public async Task SnapshotMetrics(Stream outputStream, CancellationToken token)
                 string metricName = PrometheusDataModel.GetPrometheusNormalizedName(metricInfo.Provider, metricInfo.Name, metricInfo.Unit);
                 string metricType = "gauge";
 
+                var keyValuePairs = from pair in metricInfo.Metadata select pair.Key + "=" + "\"" + pair.Value + "\"";
```

Comment:
Wanted to get people's opinions on this - limiting characters for the keys hasn't been an issue using the existing `NormalizeString` method (other than the quirk of potentially having two distinct keys become identical when replacing illegal characters with underscores). However, for the values (according to https://prometheus.io/docs/instrumenting/exposition_formats/), I'm concerned about cases where there might be some ambiguity about how the users do their formatting. Specifically, if a user wants a newline, their value would include `\n`; however, it seems that the expectation is that users aren't necessarily going to escape other things (like a single `\`), hence the need to do cleaning. 

**Example**:
Tag Value: naughty\nice
Interpretation 1: naughty(newline)ice, which should be left as `naughty\nice`
Interpretation 2: naughty(slash)nice, which should be transformed into `naughty\\nice`

There are some other edge cases like that since we don't know what kind of escaping a user will/won't have already done...unless I'm overlooking something, which is entirely possible.

Reply:
I don't think there is any way we can reason about the difference between the interpretations in the example provided. I think it's best that we just take the value, escape it according to the Prometheus docs, and write it out.